### PR TITLE
Issue #13809: Kill mutation in Checker

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13809: Kill mutation in Checker

class :- https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

------------

# Mutation
https://github.com/checkstyle/checkstyle/blob/8dffc49e46490cbd472548cfd6cccccd5ecedc4f/config/pitest-suppressions/pitest-common-suppressions.xml#L3-L10
and 
https://github.com/checkstyle/checkstyle/blob/8dffc49e46490cbd472548cfd6cccccd5ecedc4f/config/pitest-suppressions/pitest-common-suppressions.xml#L48-L55

---------

# Explaination:- 

we initialize fileExtensions to null if extensions are null at 
https://github.com/checkstyle/checkstyle/blob/8dffc49e46490cbd472548cfd6cccccd5ecedc4f/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L549-L552
so if we remove it and remove the empty initialization at 
https://github.com/checkstyle/checkstyle/blob/8dffc49e46490cbd472548cfd6cccccd5ecedc4f/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L111 
as this is not final it will be considered as null. so no issue will be there.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/Checker.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public class Checker extends AbstractAutomaticBean implements MessageDispatcher,
     private Context childContext;
 
     /** The file extensions that are accepted. */
-    private String[] fileExtensions = CommonUtil.EMPTY_STRING_ARRAY;
+    private String[] fileExtensions;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Checker.java:111]: Null is ok by this utility
https://github.com/checkstyle/checkstyle/blob/223ed3981ebf088c005dc7c8bf2d075194f77ae1/src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java#L120-L124

**Review by romani — APPROVED** [review]
Ok to merge.

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
rebased

**Review by Vyom-Yadav — APPROVED** [review]
lgtm!

**Pull request merged**